webpush-subscription-2022.bs

```diff
@@ -0,0 +1,360 @@
+</li>
+<li> 
+**Establish Subscription:**
+The *subscriber* subscribes to the *browser messaging service* to receive Web Push notifications using the `vapidPublicKey` of the *subscription service*.
+In return, the *subscriber* retrieves the `endpoint`, `auth` and `p256dh` values. 
+A corresponding subscription is submitted to the subscription endpoint.
+The *subscription service* authenticates the *subscriber* with the *Authorization Server*, checks authorization of the *subscriber* and registers the subscription.
+</li>
+<li>
+**Deliver Notifications:**
+The *subscription service* issues Push notifications to the *browser messaging serivce* which in turn delivers the Push notification to the *subscriber*.
+For each notification, the *subscription service* checks the authorization of the *subscriber* with the *Authorization Server*.
+</li>
+<li>
+**Cancel Subscription:**
+When the *subscriber* chooses not to receive Web Push notifications anymore, it unsubscribes from the *browser messaging service*. 
+Additionally, it sends an unsubscription request to the *subscription service*.
+</li>
+</ul>
+
+<div class="advisement"> 
+<b>
+It is not yet specified by [[!SOLID-NOTIFICATIONS]], how unsubscription works. Depends on #36. 
+</b>
+<ul>
+<li>
+Option (1): `as:Undo` via HTTP POST to subscription endpoint.
+</li>
+<li>
+Option (2): HTTP DELETE on a subscription.
+</li>
+</ul>
+</div>
+
+# WebPushSubscription2022 Type # {#subscription-type}
+
+This specification defines the WebPushSubscription2022 type for use with Solid Notifications subscriptions.
+The URI of the subscription type is &lt;http://www.w3.org/ns/solid/notification#WebPushSubscription2022&gt;.
+
+An WebPushSubscription2022 API MUST conform to the [Solid Notifications Protocol](https://solid.github.io/notifications/protocol#discovery).
+
+An WebPushSubscription2022 API SHOULD support the [Solid Notifications Features](https://solid.github.io/notifications/protocol#notification-features).
+
+Note: Let the class `WebPushSubscription2022` be an `rdfs:subClassOf` of `as:Follow`?
+
+The WebPushSubscription2022 type defines the following properties:
+
+: vapidPublicKey
+:: The `vapidPublicKey` property indicates the notification server's public key as defined by [[!RFC8292]], which can be used by the client for the Voluntary Application Server Identification (VAPID).
+
+%% : object
+%% :: The `object` property indicates the information resource which the client would like to receive notifications about. 
+%%    The value of source property MUST be a URI, using the `https` scheme, and identifying an information resource.
+
+%% Note: Let the predicate `topic` be an `rdfs:subClassOf` of `as:object`?
+
+%% Issue(62):
+
+: endpoint
+:: The `endpoint` property indicates the "Push Endpoint" as defined in the [[!PUSH-API]] specification.
```

Comment:
The Push API, does refer to this as an "endpoint", but in our context, `subscription` is likely the better term

Reply:
Hm. I am hesitant to call the property 'subscription' / `pushSubscription`.
- the property name is already defined by the Push API
- the property is also called `endpoint` in the corresponding JS implementation of the Push API. 
I think renaming this property would make it more confusing to an implementer.

On the other hand,
> A [push subscription](https://www.w3.org/TR/push-api/#dfn-push-subscription) has an associated push endpoint. It MUST be the absolute URL exposed by the [push service](https://www.w3.org/TR/push-api/#dfn-push-service) where the [application server](https://www.w3.org/TR/push-api/#dfn-application-server) can send [push messages](https://www.w3.org/TR/push-api/#dfn-push-message) to. A [push endpoint](https://www.w3.org/TR/push-api/#dfn-push-endpoint) MUST uniquely identify the push subscription.

indicates that the `endpoint` is quite the overloaded term when they try to say that the URL identifies the push subscription.

If I had to switch from `endpoint` to something else, I'd prefer  `pushSubscription` over `subscription` just to be explicit which one is meant.

Reply:
Since this is a property on the subscription request, not the subscription response I would not use `subscription` here. I think other subscription types call this endpoint a `target, since the publisher of notifications, should deliver them to that `target` endpoint.